File: java/1010. Pairs of Songs With Total Durations Divisible by 60/Solution.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        // we can use the fact that if time[i] % 60 = x
        // we just need to find numbers where time[j] % 60 = 60 - x

        int[] reduced = new int[60];
        int count = 0;

        for (int duration : time) {
            int remainder = duration % 60;
            if (remainder == 0) {
                count += reduced[0];
            } else {
                count += reduced[60 - remainder];
            }
            reduced[remainder]++;
        }

        return count;
    }
}

File: python/1010. Pairs of Songs With Total Durations Divisible by 60/solution.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        reduced = [0] * 60
        count = 0
        for duration in time:
            remainder = duration % 60
            if remainder == 0 :
                count += reduced[0]
            else :
                count += reduced[60 - remainder]

            reduced[remainder] += 1
        return count